### PR TITLE
Fix finding variable definitions in workspace

### PR DESCRIPTION
**Pull request by kapilmb**
Resolves PowerShell/vscode-powershell#710

### CODE REVIEW

File: src/PowerShellEditorServices/Language/FindDeclartionVisitor.cs [hunk 1]
```diff
@@ -14,25 +14,31 @@ namespace Microsoft.PowerShell.EditorServices
     internal class FindDeclartionVisitor : AstVisitor
```
Comment (daviwil) [inline src/PowerShellEditorServices/Language/FindDeclartionVisitor.cs]: Mind fixing the typo in the name of this class? Find*Declaration*Visitor.  That's bugged me for a while :)

**Comment by kapilmb** [comment]
To take `scope` into account we have two options - 
1. implement our own ad-hoc logic OR
2. use the [*variable analysis*](https://github.com/PowerShell/PSScriptAnalyzer/blob/development/Engine/VariableAnalysis.cs) part in PSSA. 

The advantage with **2** is that we will have a robust solution and the techniques used there can then also be used in *rename* logic.

**Comment by daviwil** [comment]
I'm all for using the existing variable analysis logic, maybe it's better to wait to do that until we can centralize our analysis and workspace logic in a reusable module

**Comment by kapilmb** [comment]
The existing *variable analysis* logic is still somewhat buggy and needs some work. But I agree that we should centralize our analysis logic then use it from there.

File: src/PowerShellEditorServices/Language/FindDeclartionVisitor.cs (hunk 1 above)
Comment (kapilmb) [inline src/PowerShellEditorServices/Language/FindDeclartionVisitor.cs]: fixed!

**Comment by daviwil** [comment]
Thanks dude!  Feel free to merge it once the tests pass

**Pull request merged**